clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3029,6 +3029,23 @@ static void handleNumSimdWorkItemsAttr(Sema &S, Decl *D,
                                                                      E);
 }
 
+// Handles stall_enable
+static void handleStallEnableAttr(Sema &S, Decl *D, const ParsedAttr &Attr) {
+  if (D->isInvalidDecl())
+    return;
+
+  if (S.LangOpts.SYCLIsHost)
+    return;
```

Comment:
Since we are not applying the attribute on host, I think we should allow it only on device.